azkaban-common/src/main/java/azkaban/executor/OnContainerizedExecutionEventListener.java

```diff
@@ -70,22 +70,44 @@ private void restartExecutableFlow(final ExecutableFlow exFlow) {
     // Update the flow options so that the flow will be not retried again by Azkaban
 
     // inherent the retry time counters
-    executableFlow.setUserDefinedRetryCount(exFlow.getUserDefinedRetryCount());
-    executableFlow.setSystemDefinedRetryCount(exFlow.getSystemDefinedRetryCount());
+    retryExFlow.setUserDefinedRetryCount(originalExFlow.getUserDefinedRetryCount());
```

Comment:
it is worthwhile to have a field indicating how many times it is retrying (e.g. "x"th retries). Plus, do we put a limit on number of times retry?

Reply:
yes we need this to count,
and the limit is in `ExecutionOption.FlowParameter` defined by key `"flow.max.retries"`.

This logic is implemented in #3268